File: 1635-number-of-good-pairs/number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        goodPair = 0
        freq = {}

        for num in nums:
            if num in freq:
                goodPair += freq[num]

            freq[num] = freq.get(num, 0) + 1

        return goodPair

File: 1635-number-of-good-pairs/number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        unordered_map<int,int> seen;
        int pairs =0;

        for (int num : nums){
            pairs+=seen[num];
            seen[num]++;
        }
        return pairs;
    }
};